modules/acknowledgements/jsx/acknowledgementsIndex.js

```diff
@@ -0,0 +1,370 @@
+ * and addAcknowledgement.
+ *
+ * @author Zaliqa Rosli
+ * @version 1.0.0
+ *
+ */
+class AcknowledgementsIndex extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      error: false,
+      isLoaded: false,
+      affiliationsOptions: {
+        douglas: 'Douglas',
+        mcgill: 'McGill',
+      },
+      degreesOptions: {
+        bachelors: 'Bachelors',
+        masters: 'Masters',
+        phd: 'PhD',
+        postdoc: 'Postdoctoral',
+        md: 'MD',
+        registeredNurse: 'Registered Nurse',
+      },
+      rolesOptions: {
+        investigators: 'Investigators',
+        projectAdministration: 'Project Administration',
+        databaseManagement: 'Database Management',
+        interviewDataCollection: 'Interview Data Collection',
+        dataAnalyses: 'Data Analyses',
+        mriAcquisition: 'MRI Acquisition',
+        dataEntry: 'Data Entry',
+        databaseProgramming: 'Database Programming',
+        imagingProcessingAndEvaluation: 'Imaging Processing and Evaluation',
+        geneticAnalysisAndBiochemicalAssays: 'Genetic Analysis and Biochemical Assays',
+        randomizationAndPharmacyAllocation: 'Randomization and Pharmacy Allocation',
+        consultants: 'Consultants',
+        lpCsfCollection: 'LP/CSF Collection',
+      },
+      showModal: false,
+    };
+
+    this.fetchData = this.fetchData.bind(this);
+    this.pickElements = this.pickElements.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+    this.openModalForm = this.openModalForm.bind(this);
+    this.closeModalForm = this.closeModalForm.bind(this);
+    this.renderCitationPolicy = this.renderCitationPolicy.bind(this);
+    this.renderAddForm = this.renderAddForm.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  pickElements(formElement, keys) {
+    let subset = {};
+    keys.forEach((key) => {
+      if (formElement.hasOwnProperty(key)) {
+        subset[key] = formElement[key];
+      }
+    });
+    return subset;
+  }
+
+  /**
+   * Store the value of the element in this.state.formData
+   *
+   * @param {string} formElement - name of the form element
+   * @param {string} value - value of the form element
+   */
+  setFormData(formElement, value) {
+    const formData = this.state.formData;
```

Comment:
@HenriRabalais pointed out to me in another PR how assigning state to a variable like this isn't good. You could do this instead:
`const formData = Object.assign({}, this.state.formData);`